internal/manifest/oci_index.go

```diff
@@ -76,6 +78,19 @@ func (index *OCI1IndexPublic) UpdateInstances(updates []ListUpdate) error {
 	return index.editInstances(editInstances)
 }
 
+func addCompressionAnnotations(compressionAlgorithm []compression.Algorithm, annotationsMap map[string]string) {
+	// TODO: This should also delete the algorithm if map already contains an algorithm and compressionAlgorithm
+	// list has a different algorithm.
```

Comment:
I don’t know how I feel about not implementing this… I guess let’s start with the “add a Zstd annotation on a Zstd copy” part identified earlier, and let’s see how that goes.

---

Here, though, please at least record. 
```suggestion
	// TODO: This should also delete the algorithm if map already contains an algorithm and compressionAlgorithm
	// list has a different algorithm. To do that, we would need to modify the callers to always provide a reliable
	// and full compressionAlghorithms list.
```

Reply:
I implemented `delete` for this, assuming callers would take care of choosing `gzip` in the algorithms list.